plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py

```diff
@@ -7,65 +7,18 @@
 from hydra.types import ObjectConf
 
 
-class QueueType(Enum):
-    auto = "auto"
+class ExecutorName(Enum):
     local = "local"
     slurm = "slurm"
 
 
 @dataclass
-class SlurmQueueConf:
-    # Params are used to configure sbatch, for more info check:
-    # https://github.com/facebookincubator/submitit/blob/master/submitit/slurm/slurm.py
-
-    # maximum time for the job in minutes
-    time: int = 60
-    # number of cpus to use for each task
-    cpus_per_task: int = 10
-    # number of gpus to use on each node
-    gpus_per_node: int = 1
-    # number of tasks to spawn on each node
-    ntasks_per_node: int = 1
-    # number of nodes to use for the job
-    nodes: int = 1
-    # memory to reserve for the job on each node, in GB
-    mem: str = "${hydra.launcher.mem_limit}GB"
-    # slurm partition to use on the cluster
-    partition: Optional[str] = None
-    # USR1 signal delay before timeout
-    signal_delay_s: int = 120
-    # name of the job
-    job_name: str = "${hydra.job.name}"
-    # Maximum number of retries on job timeout.
-    # Change this only after you confirmed your code can handle re-submission
-    # by properly resuming from the latest stored checkpoint.
-    # check the following for more info on slurm_max_num_timeout
-    # https://github.com/facebookincubator/submitit/blob/master/docs/checkpointing.md
-    max_num_timeout: int = 0
-
-
-@dataclass
-class LocalQueueConf:
-    # local executor mocks the behavior of slurm locally
-
-    # maximum time for the job in minutes
-    timeout_min: int = 60
-    # number of gpus to use on each node
-    gpus_per_node: int = 1
-    # number of tasks to spawn on each node (only one node available in local executor)
-    tasks_per_node: int = 1
+class BaseParams:
+    """Configuration shared by all executors
+    """
 
-
-@dataclass
-class AutoQueueConf:
-    # auto executor automatically identifies and uses available cluster
-    # Currently this is only slurm, but local executor can be manually forced
-    # instead.
-    # Most parameters are shared between clusters, some can be cluster specific
-
-    # cluster to use (currently either "slurm" or "local" are supported,
-    # None defaults to an available cluster)
-    cluster: str = "slurm"
+    executor: ExecutorName = ExecutorName.slurm  # can we get rid of this?
```

Comment:
Is this needed here? Is there another way to get this information since we now call `submitit_local` or `submitit_slurm`. That would prevent a user from setting it to the wrong value.

Reply:
Right now you are using this as the flag to gate slurm or local initialization logic.
An alternative I think will be much cleaner is to use a different class. I will comment below.